devito/ops/operator.py

```diff
@@ -24,6 +25,15 @@ def _specialize_iet(self, iet, **kwargs):
         ops_timing = Call(namespace['ops_timing_output'], [FunctionPointer("stdout")])
         ops_exit = Call(namespace['ops_exit'])
 
+        dims = []
+        for n, (section, trees) in enumerate(find_affine_trees(iet).items()):
+            dims.append(len(trees[n].dimensions))
+
+        assert (d == dims[0] for d in dims), "The OPS backend currently assumes that all kernels have the same number of dimension"
+        
+        self._includes.append('stdio.h')
+        self._headers.append('#define OPS_%sD' % dims[0])
```

Comment:
I think it would be nice if we can use a lambda expression here:

1) Define in utils.h a dictionary entry as:
namespace['ops-define-dimension'] = lambda i: '#define OPS_%sD' % i

2) Replace this line to:
self._headers.append(namespace['ops-define-dimension'](dims[0]))

Reply:
thank you ! fixed